src/status_im2/contexts/wallet/receive/view.cljs

```diff
@@ -0,0 +1,102 @@
+    [status-im2.constants :as constants]
+    [status-im2.contexts.wallet.common.sheets.network-preferences.view :as network-preferences]
+    [status-im2.contexts.wallet.receive.style :as style]
+    [utils.i18n :as i18n]
+    [utils.image-server :as image-server]
+    [utils.re-frame :as rf]))
+
+(def id-to-network
+  {constants/mainnet-chain-id  :ethereum
+   constants/optimism-chain-id :optimism
+   constants/arbitrum-chain-id :arbitrum})
+
+(defn- share-action
+  [address share-title]
+  (share/open
+   (if platform/ios?
+     {:activityItemSources [{:placeholderItem {:type    "text"
+                                               :content address}
+                             :item            {:default {:type "text"
+                                                         :content
+                                                         address}}
+                             :linkMetadata    {:title share-title}}]}
+     {:title   share-title
+      :subject share-title
+      :message address})))
+
+(defn- open-preferences
+  [selected-networks]
+  (rf/dispatch [:show-bottom-sheet
+                {:content
+                 (fn []
+                   [network-preferences/view
+                    {:blur?             true
+                     :selected-networks (set @selected-networks)
+                     :on-save           (fn [chain-ids]
+                                          (rf/dispatch [:hide-bottom-sheet])
+                                          (reset! selected-networks (map #(get id-to-network %)
+                                                                         chain-ids))
+                                        )}])}]))
+
+
+(defn view
+  []
+  (let [padding-top       (:top (safe-area/get-insets))
+        wallet-type       (reagent/atom :wallet-legacy)
+        selected-networks (reagent/atom [:ethereum :optimism :arbitrum])]
+    (fn []
+      (let [{:keys [address color emoji] :as account} (rf/sub [:wallet/current-viewing-account])
+            share-title                               (str (:name account) " " (i18n/label :t/address))
+            qr-url                                    (if (= @wallet-type :wallet-multichain)
+                                                        (as-> @selected-networks $
+                                                          (map qr-codes/get-network-short-name-url $)
+                                                          (apply str $)
+                                                          (str $ address))
+                                                        address)
+            qr-media-server-uri                       (image-server/get-qr-image-uri-for-any-url
+                                                       {:url         qr-url
+                                                        :port        (rf/sub [:mediaserver/port])
+                                                        :qr-size     500
```

Comment:
where do we get 500 from? should this be a constant at least?

Reply:
@J-Son89 This is qr-size of the QR to be scanned. I checked other usages of `image-server/get-qr-image-uri-for-any-url` and they use a QR of 500 or 600. Didn't find any constant. In the designs the size for this particular QR is 311, but that makes it look a bit blurry. Giving it a little higher value fixes that. Maybe I can open an issue to unify the `qr-size` usage for this function.

Reply:
Yeah that's fine but probably best to put it in a constant, at least in this file 👍